odoo_module_migrate/migration_scripts/migrate_130_140.py

```diff
@@ -1,12 +1,86 @@
 # Copyright (C) 2020 - Today: Iván Todorovich
+# Copyright (C) 2020 - Today: Simone Rubino
 # @author: Iván Todorovich (https://twitter.com/ivantodorovich)
+# @author: Simone Rubino (https://github.com/Daemo00)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from pathlib import Path
+import lxml.etree as et
 
-_TEXT_ERRORS = {
-    ".xml": {
-        "<act_window(\n|.|\t)*":
-        "act_window tag has been deprecated. Use <record> instead",
-        "<report(\n|.|\t)*":
-        "report tag has been deprecated. Use <record> instead",
-    }
-}
+
+def _reformat_file(file_path: Path):
+    """Reformat `file_path`.
+
+    Substitute `act_window` and `report` tag with `record` tag.
+    Note that:
+    - `id` attribute is kept in the `record` tag;
+    - in `report` tag:
+      - `string` has been renamed to `name`;
+      - `name` has been renamed to `report_name`;
+    - other attributes are assigned to respective fields
+    """
+    file_path = str(file_path.resolve())
+    parser = et.XMLParser(remove_blank_text=True)
+    tree = et.parse(file_path, parser)
+    root = tree.getroot()
+    reformat_tags = (*root.findall("act_window"), *root.findall("report"))
+    if not reformat_tags:
+        return None
+
+    for tag in reformat_tags:
+        for attrib, value in tag.attrib.items():
+            if attrib == "id":
+                continue
+            tag.attrib.pop(attrib)
+
+            if tag.tag == "report":
+                if attrib == "name":
+                    attrib = "report_name"
+                if attrib == "string":
+                    attrib = "name"
+
+            field_elem = et.SubElement(tag, "field", {"name": attrib})
+            field_elem.text = value
+
+        tag.attrib["model"] = "ir.actions." + tag.tag
+        tag.tag = "record"
+    tree.write(file_path, pretty_print=True,
+               xml_declaration=True, encoding=tree.docinfo.encoding)
```

Comment:
This overrides whole xml file. Would be great if you find a way to avoid this. You can see [on this example](https://github.com/itpp-labs/sync-addons/pull/213/commits/5412ad4fbb286aa9e85cfbfadb498d888d9abc77) how many lines are changed without a need.

Reply:
Thanks for checking and for the example, I understand the need to keep the changes as little as possible.

A way to edit just part of the file might be with regular expressions, but there are some operations that can turn out to be quite complex with a regex and I think that doing them with code is much more readable.

I have made some research but found no way to reformat an XML just partially, or reformat keeping white spaces any suggestion?

The root cause might be the `remove_blank_text` option given to the XML parser, but I added it to resolve another comment of yours in https://github.com/OCA/odoo-module-migrator/pull/38#discussion_r519045618 because otherwise the XML was not formatted at all.

Reply:
A solution could be replace whole act_window node with a node generated by lxml lib, something like

```
"<act_window[^/]*id=\"%s\"[^/]*/>" % tag["id"]
```

https://regex101.com/r/d8A77f/1

Reply:
I don't get how I should use this regular expression in the migration code, maybe it is faster if you add a commit to this PR implementing the change?
Or could you add suggestions in review comments that explain how to use that?

Reply:
So, instead of ``` tree.write(file_path, pretty_print=True, ....```
it could be something like
```
regexp = "<act_window[^/]*id=\"%s\"[^/]*/>" % tag["id"]
new_tag_str = etree.tostring(tag, pretty_print=True) 

# Read in the file
with open(file_path, 'r') as file :
  filedata = file.read()

# Replace the target string
filedata = re.sub(regexp, new_tag_str, refiledata)

# Write the file out again
with open(file_path, 'w') as file:
  file.write(filedata)
```
it's not perfect for indent, but that can be fixed manually by adding spaces into new_tag_str, e.g.

```
new_tag_str = "\n".join([" "*4 + line for line in new_tag_str.split("\n"))
```

Reply:
Thanks, I added a commit that uses a regex as you suggested, can you check if it is better now?